### DISCUSSION TITLE
Implicit wait

### DISCUSSION THREAD

**Discussion by ghost** [seleniumbase/SeleniumBase#2709] [Q&A] [answered]
Is there a proper way to do nothing on a page and let the browser run? I don't want the thread to sleep as the browser is running some javascript and I want to wait a couple of seconds to ensure it's done.

**Comment by mdmintz** [comment, marked as answer]
All waits from methods are explicit, with default timeouts set if they are not specified during the method call.
See [SeleniumBase/help_docs/method_summary.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/method_summary.md) for the methods.

Know what exactly you want to wait for, and then call the appropriate method for that.
Otherwise you would be using `sleep(seconds)` for general waiting.

There are some general waiting methods, such as `wait_for_ready_state_complete()`, and others. See the API.
